mobile-app/app/hooks/useShowAdvertisement.ts

```diff
@@ -13,20 +13,27 @@ export function useShowAdvertisement (): { showAd: boolean, counter: number, adU
   const [showAd, setShowAd] = useState(true)
   const [counter, setCounter] = useState(0)
   const [adUrl, setAdUrl] = useState<string | undefined>()
+  const [intervalID, setIntervalID] = useState<NodeJS.Timer>() // created a useState for intervalID
 
   useEffect(() => {
-    const timer = setInterval(() => {
+    setIntervalID(setInterval(() => {
       setCounter((c) => {
-        if (c - 1 === 0) {
+        if (c - 1 === 0 || c < 0) {
```

Comment:
Is this still needed? Seems to be handled below?

Reply:
ok sure

Honestly I don't even get why the clearing the timer didn't work before (or why it didn't trigger a re-redner of the surrounding view before) 🤷‍♂️, so this solution at least worked 😶‍🌫️

Reply:
Weird 🤔